src/query/sql/src/executor/physical_plan_builder.rs

```diff
@@ -278,8 +281,45 @@ impl PhysicalPlanBuilder {
             RelOperator::Join(join) => {
                 let build_side = self.build(s_expr.child(1)?).await?;
                 let probe_side = self.build(s_expr.child(0)?).await?;
-                let build_schema = build_side.output_schema()?;
-                let probe_schema = probe_side.output_schema()?;
+
+                let build_schema = match join.join_type {
+                    JoinType::Left | JoinType::Full => {
+                        let build_schema = build_side.output_schema()?;
+                        // Wrap nullable type for columns in build side.
+                        let build_schema = DataSchemaRefExt::create(
+                            build_schema
+                                .fields()
+                                .iter()
+                                .map(|field| {
+                                    DataField::new(field.name(), field.data_type().wrap_nullable())
+                                })
+                                .collect::<Vec<_>>(),
+                        );
+                        build_schema
+                    }
+
+                    _ => build_side.output_schema()?,
+                };
+
+                let probe_schema = match join.join_type {
+                    JoinType::Right | JoinType::Full => {
+                        let probe_schema = probe_side.output_schema()?;
+                        // Wrap nullable type for columns in probe side.
+                        let probe_schema = DataSchemaRefExt::create(
+                            probe_schema
+                                .fields()
+                                .iter()
+                                .map(|field| {
+                                    DataField::new(field.name(), field.data_type().wrap_nullable())
+                                })
+                                .collect::<Vec<_>>(),
+                        );
+                        probe_schema
+                    }
+
+                    _ => probe_side.output_schema()?,
+                };
+
```

Comment:
@andylokandy  This will solve the problem that it's hard to measure the nullability of columns with outer joins.